### PR TITLE
Minor fixes and refactoring.

### LINKED ISSUE #411: Fix logging of progress in HetPulldownCalculator.
Current implementation, which is supposed to print messages after multiples of 5% of the sites have been traversed, is a bit hacky and possibly incorrect when loci are uncovered (mea culpa!).  Example:

    13:50:52.653 INFO  HetPulldownCalculator - Examining 5398749 sites...
    14:14:01.229 INFO  HetPulldownCalculator - Examined 269937 out of 5398749 sites.
    14:17:11.393 INFO  HetPulldownCalculator - Examined 5398749 sites.

We see that only the first message was output.  Low priority, but I think this is an easy fix---will investigate.

### LINKED ISSUE #417: Change pulldown argument for PlotACNVResults CLI
PlotACNVResults takes as one of its inputs the pulldown produced by GetHetCoverage.  Currently the argument for this is -SNP; however, this is inconsistent with the use of -SNP to refer to the list of common SNPs in GetHetCoverage.

The GetHetCoverage arguments referring to the pulldowns are -NHET and -THET for the normal and tumor, respectively.  So perhaps this should be -HET?

### LINKED ISSUE #434: Extract checks for missing columns in table readers.
from a PR comment by @davidbenjamin:

    try (final TableReader<AllelicCount> reader = TableUtils.reader(inputFile,
        (columns, formatExceptionFactory) -> {
            if (!columns.containsAll(AllelicCountTableColumns.COLUMN_NAME_ARRAY)) {

This pattern appears in several places. Why not extract a utility method that checks for all required columns and throws an error with the missing columns? This method would have a signature like checkRequiredColumns(final TableColumnCollection columns, final Function<String, RuntimeException> formatExceptionFactory, final Collection<String> requiredColumnNames). Honestly if it were up to me I'd go a little further and replace arbitrarily-typed formatExceptionFactory in TableReader with some fixed exception, eg MissingColumnsException, but that's beyond the scope of this PR.

Places where this refactoring could simplify code:

CoverageDropoutResult.java line 172
SegmentUtils.java line 426
TargetTableReader.java line 51
SampleCoverageStatsReader.java line 46

### PR DESCRIPTION
**Pull request by samuelklee**
Closes #411, closes #417, closes #434.

### CODE REVIEW

**Comment by samuelklee** [comment]
@davidbenjamin please feel free to comment as well.

File: src/test/java/org/broadinstitute/hellbender/tools/exome/PlotACNVResultsIntegrationTest.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
     public void testACNVPlotting() {
         File tDir = IOUtils.tempDir("Test", "Plotting");
         final String[] arguments = {
-                "-" + ExomeStandardArgumentDefinitions.SNP_FILE_SHORT_NAME, SNP_COUNTS_FILE.getAbsolutePath(),
+                "-" + ExomeStandardArgumentDefinitions.ALLELIC_COUNTS_FILE_SHORT_NAME, SNP_COUNTS_FILE.getAbsolutePath(),
```
Comment (LeeTL1220) [inline src/test/java/org/broadinstitute/hellbender/tools/exome/PlotACNVResultsIntegrationTest.java]: What is SNP_COUNTS_FILE in this context?

**Comment by LeeTL1220** [comment]
@samuelklee Just a question, otherwise, this is pretty straightforward.

File: src/test/java/org/broadinstitute/hellbender/tools/exome/PlotACNVResultsIntegrationTest.java (hunk 1 above)
Comment (samuelklee) [inline src/test/java/org/broadinstitute/hellbender/tools/exome/PlotACNVResultsIntegrationTest.java]: it's a simulated tumor pulldown.  i went ahead and renamed it to TUMOR_ALLELIC_COUNTS_FILE throughout.

File: src/main/java/org/broadinstitute/hellbender/tools/exome/TargetCoverageUtils.java [hunk 2]
```diff
@@ -30,16 +30,16 @@
...
     /**
      * Function to read target coverage.
      */
     private static final BiFunction<TableColumnCollection, Function<String, RuntimeException>, Function<DataLine, TargetCoverage>> tableColumnCollectionFunctionFunctionBiFunction = (columns, formatExceptionFactory) -> {
-        if (!columns.containsAll(TARGET_NAME_COLUMN, CONTIG_COLUMN, START_COLUMN, END_COLUMN) || (columns.columnCount() < 5))
-            throw formatExceptionFactory.apply("Bad header");
-        if (columns.columnCount() > 5)
-            throw formatExceptionFactory.apply("Bad header -- more than one sample included in the input file.");
+        TableUtils.checkMandatoryColumns(columns, TARGET_COVERAGE_COLUMN_NAME_ARRAY, formatExceptionFactory);
+        if (columns.columnCount() != 5) {
```
Comment (davidbenjamin) [inline src/main/java/org/broadinstitute/hellbender/tools/exome/TargetCoverageUtils.java]: 5 is a magic number.  How about `TARGET_COVERAGE_COLUMN_NAME_ARRAY.length + 1`?

**Comment by davidbenjamin** [comment]
Just one quibble from me.  I commend your adherence to the Boy Scout Rule.

**Comment by samuelklee** [comment]
thanks @davidbenjamin.  took a peek in your coverage branch and it looks like you might have made some additional sensible changes to TableUtils too---will let you add those in that branch!

**Pull request merged**